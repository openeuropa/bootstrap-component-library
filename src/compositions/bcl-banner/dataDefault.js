const drupalAttribute = require("drupal-attribute");

module.exports = {
  title: "EU Budget for the future",
  description: "Innovation, economy, environment and geopolitics",
  link: {
    label: "Subscribe",
    path: "/example.html",
    icon: {
      name: "chevron-right",
      path: "boostrap-icons.svg",
    },
    attributes: new drupalAttribute().addClass(["btn", "btn-primary"]),
  },
  attributes: new drupalAttribute().addClass([
    "bg-light",
    "text-dark",
    "text-center",
  ]),
};
