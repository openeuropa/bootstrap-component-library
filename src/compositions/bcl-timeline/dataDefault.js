const drupalAttribute = require("drupal-attribute");

module.exports = {
  id: 1,
  toggle_collapsed: "Show %d more items",
  toggle_expanded: "Hide %d items",
  hide: {
    from: 5,
    to: -2,
  },
  icons_path: "boostrap-icons.svg",
  items: [
    {
      label: "13 September 2017",
      title: "Item title",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "28-29 September 2017",
      content: '<a href="/example">Informal Digital Summit, Tallinn</a>',
    },
    {
      label: "14 November 2017",
      content:
        "Strengthening European identity through education and culture: European Commission's contribution to the Leaders' meeting.",
    },
    {
      label: "17 November 2017",
      content: '<a href="/example">Social Summit in Gothenburg, Sweden</a>',
    },
    {
      label: "6 December 2017",
      content:
        '<a href="/example">Economic and Monetary Union package of proposals</a>',
    },
    {
      label: "14-15 December 2017",
      content:
        '<a href="/example">EU Leaders\' meeting on migration, Brussels</a>',
    },
    {
      label: "15 December 2017",
      content: '<a href="/example">Euro Summit</a>',
    },
    {
      label: "6 February 2018",
      content: '<a href="/example">EU-Western Balkans Strategy</a>',
    },
    {
      label: "14 February 2018",
      content:
        '<a href="/example">Multiannual Financial Framework</a> and <a href="/example">institutional issues</a> - enhancing efficiency at the helm of the European Union',
    },
    {
      label: "23 February 2018",
      content: '<a href="/example">Informal European Council</a>',
    },
    {
      label: "22-23 March 2018",
      content: '<a href="/example">European Council</a>',
    },
    {
      label: "2 May 2018",
      content:
        '<a href="/example">Commission proposal: Long-term budget post-2020</a>',
    },
    {
      label: "15 May 2018",
      content:
        '<a href="/example">European Commissionâ€™s contribution to the Informal Leaders\' meeting</a>, Sofia, Bulgaria (16 May 2018)',
    },
    {
      label: "17 May 2018",
      content:
        '<a href="/example">EU-Western Balkans Summit</a> (Sofia, Bulgaria)',
    },
    {
      label: "23 May 2018",
      content:
        'Progress by Member States in meeting the conditions for adopting the euro - <a href="/example">reports</a> by the European Commission and ECB',
    },
    {
      label: "23 May 2018",
      content:
        '<a href="/example">European Semester</a>: European Commission publishes economic policy guidance for Member States',
    },
    {
      label: "29 May - 14 June 2018",
      content:
        '<a href="/example">EU budget for the future</a>: European Commission\'s legislative proposals for spending programmes',
    },
    {
      label: "28-29 June 2018",
      content:
        'Commission contribution to the European Council: <a href="/example">Economic and Monetary Union, migration, EU budget for the future</a>',
    },
    {
      label: "10 July 2018",
      content:
        'Ways to enhance subsidiarity, proportionality and better regulation in the daily operations of the EU: <a href="/example">Report</a> by the Task Force chaired by Commission First Vice-President Timmermans',
    },
    {
      label: "12 September 2018",
      content:
        '<a href="/example">President Juncker\'s State of the Union speech</a>',
    },
    {
      label: "20 September 2018",
      content:
        '<a href="/example">Informal European Council/Meeting of EU27 (Salzburg, Austria)</a>',
    },
    {
      label: "18 October 2018",
      content: '<a href="/example">European Council meeting</a>, Brussels',
    },
    {
      label: "21 November 2018",
      content:
        '<a href="/example">European Semester</a>: setting the economic and social priorities',
    },
    {
      label: "28 November 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">Capital Markets Union and Banking Union</a>',
    },
    {
      label: "04 December 2018",
      content:
        'Commission contribution to the European Council on <a href="/example">long-term EU budget</a> and <a href="/example">migration</a>',
    },
    {
      label: "05 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">international role of the euro</a> and the <a href="/example">Action Plan on disinformation</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the European Council on the <a href="/example">Citizens\' Dialogues</a>',
    },
    {
      label: "11 December 2018",
      content:
        'Commission contribution to the <a href="/example">Euro Summit</a>',
    },
    {
      label: "13-14 December 2018",
      title: "Item title",
      content: "European Council and Euro Summit",
    },
  ],
  attributes: new drupalAttribute(),
};
