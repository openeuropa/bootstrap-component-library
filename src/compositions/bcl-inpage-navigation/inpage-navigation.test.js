import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import dataDefault from "@openeuropa/bcl-inpage-navigation/dataDefault.js";

const template = "@oe-bcl/bcl-inpage-navigation/inpage-navigation.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Inpage navigation", () => {
  test("default renders correctly", () => {
    expect.assertions(1);

    return expect(render(dataDefault)).resolves.toMatchSnapshot();
  });
});
