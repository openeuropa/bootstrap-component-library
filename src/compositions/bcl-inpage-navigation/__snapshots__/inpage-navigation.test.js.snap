// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Inpage navigation default renders correctly 1`] = `
<jest>
  <div
    class="row"
  >
    <div
      class="col-md-3 d-none d-md-block"
    >
      <nav
        class="position-sticky"
        id="bcl-inpage-navigation"
      >
        <h5>
          Page content
        </h5>
        <ul
          class="nav nav-pills flex-column"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#item-1"
            >
              Heading 1
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#item-2"
            >
              Heading 2 with a long title going on several lines
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#item-3"
            >
              Heading 3
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#item-4"
            >
              Heading 4
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="col-md-9"
    >
      <h2
        id="item-1"
      >
        Heading 1
      </h2>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <h2
        id="item-2"
      >
        Heading 2 with a long title going on several lines
      </h2>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <h2
        id="item-3"
      >
        Heading 3
      </h2>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <h2
        id="item-4"
      >
        Heading 4
      </h2>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
      <p>
        Et pariatur qui quis ullamco magna esse aliqua. Ea eiusmod commodo in laborum nulla cillum anim. Culpa sint dolor aliquip pariatur proident ipsum id. Est voluptate et eiusmod labore ea ad consectetur minim nostrud ad. Exercitation sint laborum velit consequat magna dolor voluptate dolore incididunt nostrud eu deserunt sunt. Mollit qui sunt commodo id voluptate voluptate Lorem dolore est voluptate mollit consectetur ullamco est. Enim minim sunt ex non pariatur ex labore qui laborum sint consequat. Aute cupidatat id sint incididunt Lorem Lorem aute aliqua sint aliquip commodo laborum quis. Ex irure sit reprehenderit in amet. Fugiat in amet proident consequat occaecat officia et consequat ipsum aute commodo. Pariatur duis consequat ipsum enim quis Lorem sint eiusmod esse. Occaecat amet esse aliquip non incididunt nisi sint do. Magna deserunt irure est nostrud deserunt in elit in enim do occaecat consequat. Incididunt duis reprehenderit commodo mollit culpa ex. Cillum ullamco cupidatat sit fugiat exercitation voluptate elit nulla duis. Dolor deserunt ad aute sint. Sunt cillum aute anim quis anim anim. Voluptate enim proident voluptate deserunt mollit. Veniam exercitation ex aliquip laborum commodo esse laboris. Veniam exercitation dolore quis ea minim magna. Labore adipisicing magna in excepteur esse ullamco aliqua tempor veniam id anim. Nulla deserunt laboris aute reprehenderit nisi quis commodo. Excepteur tempor nulla amet fugiat ullamco sit. Nostrud proident exercitation veniam elit ullamco ullamco officia minim officia commodo id duis. Laboris ullamco aliquip voluptate dolore aliquip nostrud magna ut cillum nisi officia.
      </p>
    </div>
  </div>
</jest>
`;
